File: interface/go/server/yock_test.go
// Copyright 2023 The Yock Authors. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package yocki

import (
	"os"
	"testing"
	"time"

	yocki "github.com/ansurfen/yock/interface/go"
)

func TestYocki(t *testing.T) {
	go func() {
		time.Sleep(100 * time.Second)
		os.Exit(0)
	}()
	s := New()
	s.Register("SayHello", func(req *yocki.CallRequest) (*yocki.CallResponse, error) {
		return &yocki.CallResponse{Buf: "I'm Golang"}, nil
	})
	s.Run(9090)
}

File: interface/c/yock_test.c
// Copyright 2023 The Yock Authors. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

#include <stdio.h>
#include "libyock/yocki.h"

CallRequest sayHello(CallRequest req)
{
    callRequest request = callRequestBuild(req);
    printf("Fn: %d, Arg: %d", request.fn, request.arg);
    return CallResponseBuild({.buf = "I'm C"});
}

int main(int argc, char **argv)
{
    YockBuilder();
    YockCall("SayHello", sayHello);
    YockRun();
    return 0;
}